apis/acmpca/v1alpha1/certificateauthority_types.go

```diff
@@ -0,0 +1,212 @@
+	// +optional
+	// +immutable
+	GenerationQualifier *string `json:"generationQualifier,omitempty"`
+
+	// Concatenation of first letter of the GivenName, Middle name and SurName.
+	// +optional
+	// +immutable
+	Initials *string `json:"initials,omitempty"`
+
+	// First name
+	// +optional
+	// +immutable
+	GivenName *string `json:"givenName,omitempty"`
+
+	// Shortened version of a longer GivenName
+	// +optional
+	// +immutable
+	Pseudonym *string `json:"pseudonym,omitempty"`
+
+	// The certificate serial number.
+	// +optional
+	// +immutable
+	SerialNumber *string `json:"serialNumber,omitempty"`
+
+	// Surname
+	// +optional
+	// +immutable
+	Surname *string `json:"surname,omitempty"`
+
+	// Title
+	// +optional
+	// +immutable
+	Title *string `json:"title,omitempty"`
+}
+
+// CertificateAuthoritySpec defines the desired state of CertificateAuthority
+type CertificateAuthoritySpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  CertificateAuthorityParameters `json:"forProvider"`
+}
+
+// CertificateAuthorityExternalStatus keeps the state of external resource
+type CertificateAuthorityExternalStatus struct {
+	// String that contains the ARN of the issued certificate Authority
+	CertificateAuthorityARN string `json:"certificateAuthorityArn"`
+
+	// Serial of the Certificate Authority
+	Serial *string `json:"serial"`
+}
+
+// An CertificateAuthorityStatus represents the observed state of an CertificateAuthority manager.
+type CertificateAuthorityStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     CertificateAuthorityExternalStatus `json:"atProvider"`
+}
+
+// CertificateAuthorityParameters defines the desired state of an AWS CertificateAuthority.
+type CertificateAuthorityParameters struct {
+	// Type of the certificate authority
+	// +kubebuilder:validation:Enum=ROOT;SUBORINATE
+	Type acmpca.CertificateAuthorityType `json:"type"`
+
+	// Status of the certificate authority
+	// +optional
+	// +kubebuilder:validation:Enum=CREATING;PENDING_CERTIFICATE;ACTIVE;DELETED;DISABLED;EXPIRED;FAILED
+	Status acmpca.CertificateAuthorityStatus `json:"status,omitempty"`
+
+	// RevocationConfiguration to associate with the certificateAuthority.
+	RevocationConfiguration RevocationConfiguration `json:"revocationConfiguration"`
+
+	// CertificateAuthorityConfiguration to associate with the certificateAuthority.
+	CertificateAuthorityConfiguration CertificateAuthorityConfiguration `json:"certificateAuthorityConfiguration"`
+
+	// Token to distinguish between calls to RequestCertificate.
+	// +optional
+	IdempotencyToken *string `json:"idempotencyToken,omitempty"`
+
+	// The number of days to make a CA restorable after it has been deleted
+	// +optional
+	PermanentDeletionTimeInDays *int64 `json:"permanentDeletionTimeInDays,omitempty"`
+
+	// One or more resource tags to associate with the certificateAuthority.
+	Tags []Tag `json:"tags,omitempty"`
```

Comment:
```suggestion
      // +optional
	Tags []Tag `json:"tags,omitempty"`
```

Reply:
Tags are required field so can not make it as optional.

Reply:
```suggestion
	Tags []Tag `json:"tags"`
```
If required then they should also not be `omitempty`